app/src/main/java/com/kickstarter/viewmodels/projectpage/AddOnsViewModel.kt

```diff
@@ -0,0 +1,207 @@
+                        )
+                    )
+                }
+            }
+            .addToDisposable(disposables)
+
+        currentUserReward
+            .filter {
+                !RewardUtils.isDigital(it) && RewardUtils.isShippable(it) && !RewardUtils.isLocalPickup(it)
+            }
+            .compose<Pair<Reward, List<ShippingRule>>>(
+                Transformers.combineLatestPair(
+                    shippingRules
+                )
+            )
+            .switchMap { getDefaultShippingRule(it.second) }
+            .subscribe {
+                defaultShippingRule.onNext(it)
+                currentShippingRule = it
+                viewModelScope.launch {
+                    mutableAddOnsUIState.emit(
+                        AddOnsUIState(
+                            currentShippingRule = it,
+                            shippingSelectorIsGone = shippingSelectorIsGone,
+                            currentAddOnsSelection = currentAddOnsSelections
+                        )
+                    )
+                }
+            }.addToDisposable(disposables)
+
+        val shippingRule = getSelectedShippingRule(defaultShippingRule, currentUserReward)
+
+        shippingRule
+            .distinctUntilChanged { rule1, rule2 ->
+                rule1.location()?.id() == rule2.location()?.id() && rule1.cost() == rule2.cost()
+            }
+            .subscribe {
+                currentShippingRule = it
+                this.shippingRuleSelected.onNext(it)
+            }
+            .addToDisposable(disposables)
+    }
+
+    private fun getDefaultShippingRule(shippingRules: List<ShippingRule>): Observable<ShippingRule> {
+        return this.currentConfig.observable()
+            .map { it.countryCode() }
+            .map { countryCode ->
+                shippingRules.firstOrNull { it.location()?.country() == countryCode }
+                    ?: shippingRules.first()
+            }
+    }
+    private fun getSelectedShippingRule(
+        defaultShippingRule: Observable<ShippingRule>,
+        reward: Observable<Reward>
+    ): Observable<ShippingRule> {
+        return Observable.combineLatest(
+            defaultShippingRule.startWith(ShippingRuleFactory.emptyShippingRule()),
+            reward
+        ) { defaultShipping, rw ->
+            return@combineLatest chooseShippingRule(defaultShipping, rw)
+        }
+    }
+    private fun chooseShippingRule(defaultShipping: ShippingRule, rw: Reward): ShippingRule =
+        when {
+            RewardUtils.isDigital(rw) || !RewardUtils.isShippable(rw) || RewardUtils.isLocalPickup(
+                rw
+            ) -> ShippingRuleFactory.emptyShippingRule()
+            // sameReward -> backingShippingRule // TODO: When changing reward for manage pledge flow
+            else -> defaultShipping
+        }
+
+    fun provideProjectData(projectData: ProjectData) {
+        // no-op
+    }
+
+    // UI events
```

Comment:
Below are the UI events this VM will handle:
- user selects a reward
- user selects a shipping location
- user adds/removes add-on
- user clicks continue button